src/threading.ts

```diff
@@ -117,12 +118,33 @@ class Thread {
     public writeTraceback(): void {
         this.traceback = "Traceback (most recent call last):\n" + this.traceback;
         process.stdout.write(this.traceback);
+        this.exit();
     }
 
-    // Output the traceback when thread cannot handle an exception, and exit the thread
-    public throwException(): void {
-        this.writeTraceback();
-        this.exit();
+    public throwException(f: Py_FrameObject): void {
```

Comment:
Why does `throwException` take a frame object? Why doesn't it take an exception object?

I would expect that `thread.throwException(obj)` would attempt to throw `obj` on the thread, beginning with the frame at the top of the stack. You'd pass it into `tryCatchException`, which would check if the frame can handle the given exception.